### ISSUE TITLE
Lint/SafeNavigationChain cop's autocorrect adds extraneous incorrect safe navigation operators

### ISSUE THREAD

**Issue by HotFusionMan** [rubocop/rubocop#11047]
The following code is incorrectly autocorrected by the -a command line option.
```ruby
    current_mortgage_balance =
      if current_mortgage_type == "FORWARD"
        -@nrc_user_data_service
          .user_data_value("EstMortgageBalance", scenario_id: @current_scenario_id)
          &.to_i
      else
        0
      end
```
becomes
```ruby
    current_mortgage_balance =
      if current_mortgage_type == "FORWARD"
        -@nrc_user_data_service
          .user_data_value("EstMortgageBalance", scenario_id: @current_scenario_id)
          &.to_i
      &.else&.
        0
      end
```

--------

## Expected behavior

Autocorrect should not add &. at the beginning and end of an `else` statement line.

## Actual behavior

See above.

## Steps to reproduce the problem

See code sample above.

## RuboCop version

> bundle exec rubocop -V
> 1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.1) [x86_64-darwin18]
>   - rubocop-performance 1.15.0
>   - rubocop-rails 2.16.1
>   - rubocop-rspec 2.13.2
